src/main/clojure/aerospike_clj/utils.clj

```diff
@@ -39,8 +40,18 @@
 
 (defn v->array
   "An optimized way to convert vectors into Java arrays of type `clazz`."
-  [clazz v]
-  (.toArray ^Collection v ^"[Ljava.lang.Object;" (make-array clazz (count v))))
+  ([clazz ^Collection v]
+   (.toArray v ^"[Ljava.lang.Object;" (make-array clazz 0)))
+  ([clazz mapper-fn ^Collection v]
+   (let [size     (.size v)
+         res      ^"[Ljava.lang.Object;" (make-array clazz size)
+         iterator (.iterator v)]
+     (loop [i 0]
+       (when (and (< i size)
+                  (.hasNext iterator))
```

Comment:
In what case can the iterator be existed while we haven't fully populated `res`? I think it can only be if `v` is not thread-safe and another thread manipulated it during the iteration.

Reply:
In most Java collections that I know, modifying a collection while iterating over will cause an undefined state, usually an exception.  
(And if it's a Clojure collection like a `Persistent*`, a `.add` method will throw an `UnsupportedOperationException` exception

```clojure
(let [lst      (ArrayList. [1 2 3])
      iterator (.iterator lst)]
  (println (.next iterator))
  (.add lst 4)
  (println (.next iterator))
  (println (.next iterator)))
1
Execution error (ConcurrentModificationException) at java.util.ArrayList$Itr/checkForComodification (ArrayList.java:1013).
null
```